_posts/2018-02-15-Rust-1.24.md

```diff
@@ -0,0 +1,128 @@
+`rustup` knows what compiler version you're using, and so is able to pair it with the correct
+`rustfmt` version. Once you've added this component to a given toolchain, `rustup` will
+automatically update it as well. Run the command once, and then on every future
+`rustup update stable`, `rustfmt-preview` will be updated as well.
+
+> This strategy is expected to be used for other developer tools that integrate tightly with
+> the compiler, like the RLS and Clippy, in the future.
+
+Finally, please take note of that `-preview` in the name: `rustfmt` is still not quite
+at 1.0 yet. Some tweaks to the default styles are still occuring, though they're relatively
+minor. Once `rustfmt` hits 1.0, we'll be releasing a `rustfmt` component and
+deprecating `rustfmt-preview`. Since this is the first major component we're distributing
+this way, we wanted to clearly signal that it is still developing.
+
+For more, please check out [`rustfmt` on GitHub](https://github.com/rust-lang-nursery/rustfmt).
+
+#### Incremental compilation
+
+Back in September, we blogged about [Incremental Compilation](https://blog.rust-lang.org/2016/09/08/incremental.html).
+While that post goes into the details, the idea is basically this: when you're working on
+a project, you often compile it, then change something small, then compile again. Historically,
+the compiler has compiled your *entire* project, no matter how little you've changed the code.
+The idea with incremental compilation is that you only need to compile the code you've actually
+changed, which means that that second build is faster.
+
+As of Rust 1.24, this is now [turned on by default](https://github.com/rust-lang/cargo/pull/4817).
+This means that your builds should get faster!
+
+This is still not the end story for compiler performance generally, nor incremental compilation
+specifically. We have a lot more work planned in the future. For example, another change
+related to performance hit stable this release:
+[`codegen-units` is now set to 16 by default](https://github.com/rust-lang/rust/pull/46910).
+More to come!
+
+#### Other good stuff
+
+There's one other change we'd like to talk about here: undefined behavior. Rust generally
+strives to minimize undefined behavior, having none of it in safe code, and as little as
+possible in unsafe code. One area where you could invoke UB is when a `panic!` goes
+across an FFI boundary. In other words, this:
+
+```rust
+extern "C" fn panic_in_ffi() {
+    panic!("Test");
+}
+```
+
+This cannot work, as the exact mechanism of how panics work would have to be reconciled
+with how the `"C"` ABI works, in this example, or any other ABI in other examples.
+
+In Rust 1.24, [this code will now abort](https://github.com/rust-lang/rust/pull/46833)
+instead of producing undefined behavior.
+
+See the [detailed release notes][notes] for more.
+
+### Library stabilizations
+
+If you're a fan of `str::find`, used to find a given `char` inside of a `&str`, you'll be
```

Comment:
Might want to put `which is` in front of used. I had to reread the sentence a few times thinking the wording was a typo